flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/ArcticCatalog.java

```diff
@@ -359,8 +349,184 @@ private static void validateFlinkTable(CatalogBaseTable table) {
 
   @Override
   public void alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean ignoreIfNotExists)
-      throws CatalogException {
-    throw new UnsupportedOperationException();
+      throws CatalogException, TableNotExistException {
+    validateFlinkTable(newTable);
+
+    TableIdentifier tableIdentifier = getTableIdentifier(tablePath);
+    ArcticTable arcticTable;
+    try {
+      arcticTable = internalCatalog.loadTable(tableIdentifier);
+    } catch (org.apache.iceberg.exceptions.NoSuchTableException e) {
```

Comment:
Maybe you can import org.apache.iceberg.exceptions.NoSuchTableException

Reply:
sure